webrender/src/resource_cache.rs

```diff
@@ -656,7 +692,15 @@ impl ResourceCache {
             tile,
         };
         let image_info = &self.cached_images.get(&key);
-        self.texture_cache.get(&image_info.texture_cache_handle)
+
+        // If an image is not in cached_images, just return an invalid CacheItem
+        // for that request. That invalid item will be skipped during the rendering.
+        //
+        // TODO(Jerry): add a debug option to fill the corresponding area for
+        // that invalid CacheItem.
+        image_info.as_ref().ok().map_or(CacheItem::invalid(), |image_info| {
```

Comment:
let's just return `Result<CacheItem, _>` from this function and match it on the other side, there is no need for nullable `CacheItem` semantic (that `invalid()` does here)